srcCpp03/RTIDDSImpl.cxx

```diff
@@ -205,6 +206,15 @@ bool RTIDDSImpl<T>::validate_input()
                 _PM->set<long>("batchSize", 0); // Disable Batching
             }
         }
+
+        if (_isFlatData) {
+            if (_PM->is_set("batchSize") && _PM->get<long>("batchSize") > 0) {
+                fprintf(stderr, "Batching cannot be used with FlatData.\n");
+                return false;
+            } else {
+                _PM->set<long>("batchSize", -3);
```

Comment:
what is -3? If it has a special meaning, use a constant variable.

Reply:
Yes, it indicates that batchSize argument is disabled because FlatData is active. I was copying this behavior from line 192
```C++
if (_isLargeData) {
            if (_PM->is_set("batchSize") && _PM->get<long>("batchSize") != 0) {
                std::cerr << "Batching cannot be used with Large Data." 
                          << std::endl;
                return false;
            } else {
                _PM->set<long>("batchSize", -2);
            }
} else ...
```

These codes are used in `perftest_publisher.cpp` to display a message in the "header" of perftest explaining the reason why the batchsize argument is disabled. 

```C++
     if (_PM.get<long>("batchSize") == -1) {
           if (_PM.get<bool>("latencyTest")) {
               stringStream << "\t\t  BatchSize disabled for a Latency Test\n";
           } else {
               stringStream << "\t\t  BatchSize is smaller than 2 times\n"
                            << "\t\t  the sample size.\n";
           }
       } else if (_PM.get<long>("batchSize") == -2) {
           stringStream << "\t\t  BatchSize cannot be used with\n"
                        << "\t\t  Large Data.\n";
        } else if (_PM.get<long>("batchSize") == -3) {
            stringStream << "\t\t  BatchSize cannot be used with\n"
                         << "\t\t  FlatData.\n";
        }
```

I will speak with @javmorcas about refactoring this.

Reply:
Create an issue for after 3.0.0 to use constant names instead of the plain numbers.